### DISCUSSION TITLE
bme280 ERROR

### DISCUSSION THREAD

**Discussion by IsaNocNoc** [micropython/micropython#12783] [RP2040 / Pico]
Hello, when I wanted to program a simple program for a bme280, an error was generated in the library and I don't know why. Could anyone explain to me why it is generated and how to solve it?

CODE:
```
from machine import Pin,I2C
from time import sleep_ms
from ssd1306 import SSD1306_I2C
import bme280
import framebuf

i2c_1 = I2C(0, scl=Pin(17), sda=Pin(16))
i2c_2 = I2C(1, scl=Pin(3), sda=Pin(2))
oled = SSD1306_I2C(128,64,i2c_1)
bme = bme280.BME280(i2c=i2c_2)

while True:
    print(bme.values)
    sleep_ms(500)
```

ERROR:
```
Traceback (most recent call last):
  File "<stdin>", line 10, in <module>
  File "/lib/bme280.py", line 75, in __init__
OSError: [Errno 5] EIO
```

**Comment by robert-hh** [comment]
What is the error?
Edit: which BME280 breakout do you use?

**Reply by IsaNocNoc** [reply]
i'm sorry, is this
```
Traceback (most recent call last):
  File "<stdin>", line 10, in <module>
  File "/lib/bme280.py", line 75, in __init__
OSError: [Errno 5] EIO
```

**Reply by robert-hh** [reply]
The EIO Error is typically caused by wrong wiring. Please check:
- that SCL and SDA are assigned to the proper ports. Please be aware, that the numbers given at scl=xx and sda=xxx are GPIO numbers, NOT board pin numbers!
- that GND and Vcc are properly connected to the BME280.
- that there are pull-up resistors at SDA and SCL.
- that you use the right address.

You can check the proper wiring with i2c_1.scan(), which should return the address of your BME280 as decimal number, which is 118 (0x76) or 119 (0x77). If i2c_1.scan() returns an empty list or a list wit many numbers (like 8..119), then this also tells that the wiring is wrong.

**Comment by bidrohini** [comment]
OSError: [Errno 5] EIO tells you that the I2C device is not responding. Something broke in the communication.

check the wiring (SDA, SCL, GND, VCC)
check if pull-up resistors are present.
try i2c.scan(). Ref: https://github.com/orgs/micropython/discussions/10666
If nothing helps, you can cross-check your BME280 with an Arduino code like this.
https://www.pcbway.com/project/shareproject/Arduino_Barometer_Barograph_with_BME280_Sensor_64e355f0.html
